File: app/src/main/java/com/rendersoncs/reportform/viewHolder/ReportListViewHolder.java
package com.rendersoncs.reportform.viewHolder;

import android.app.AlertDialog;
import android.content.Context;
import android.view.View;
import android.widget.ImageView;
import android.widget.PopupMenu;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.rendersoncs.reportform.R;
import com.rendersoncs.reportform.constants.ReportConstants;
import com.rendersoncs.reportform.extension.StringExtension;
import com.rendersoncs.reportform.itens.ReportItems;
import com.rendersoncs.reportform.listener.OnInteractionListener;

import java.lang.reflect.Field;
import java.lang.reflect.Method;

public class ReportListViewHolder extends RecyclerView.ViewHolder {

    private TextView CompanyView;
    private TextView DateView;
    private ImageView overflow;
    private Context mContext;

    public ReportListViewHolder(@NonNull View itemView, Context context) {
        super(itemView);

        this.CompanyView = itemView.findViewById(R.id.companyView);
        this.DateView = itemView.findViewById(R.id.dateView);
        this.overflow = itemView.findViewById(R.id.overflow);
        this.mContext = context;
    }

    public void bindData(final ReportItems repoEntity, final OnInteractionListener listener) {
        this.CompanyView.setText(StringExtension.limitsText(repoEntity.getCompany(), ReportConstants.CHARACTERS.LIMITS_TEXT));
        this.DateView.setText(repoEntity.getDate());

        itemView.setOnClickListener(v -> listener.onOpenPdf(repoEntity.getId()));

        this.overflow.setOnClickListener(v -> {

            PopupMenu popupMenu = new PopupMenu(mContext, overflow);
            setForceShowIcon(popupMenu);
            popupMenu.inflate(R.menu.menu_main);
            popupMenu.setOnMenuItemClickListener(item -> {
                switch (item.getItemId()) {
                    case R.id.action_resume:
                        listener.onListClick(repoEntity.getId());
                        return true;

                    case R.id.action_edit:
                        listener.onEditReport(repoEntity.getId());
                        return true;

                    case R.id.action_share:
                        listener.onShareReport(repoEntity.getId());
                        return true;

                    case R.id.action_remove:
                        new AlertDialog.Builder(mContext)
                                .setTitle(R.string.alert_remove_report)
                                .setMessage(R.string.alert_remove_report_text)
                                .setPositiveButton(R.string.confirm, (dialog, which) -> {
                                    // Delete item
                                    listener.onDeleteClick(repoEntity.getId());
                                })
                                .setNeutralButton(R.string.cancel, null)
                                .show();
                        return true;

                    default:
                        return false;
                }
            });
            popupMenu.show();
        });

    }
    private void setForceShowIcon(PopupMenu popupMenu) {
        try {
            Field[] mFields = popupMenu.getClass().getDeclaredFields();
            for (Field field : mFields) {
                if ("mPopup".equals(field.getName())) {
                    field.setAccessible(true);
                    Object menuPopupHelper = field.get(popupMenu);
                    assert menuPopupHelper != null;
                    Class<?> popupHelper = Class.forName(menuPopupHelper.getClass().getName());
                    Method mMethods = popupHelper.getMethod("setForceShowIcon", boolean.class);
                    mMethods.invoke(menuPopupHelper, true);
                    break;
                }
            }
        } catch (Throwable e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/rendersoncs/reportform/view/adapter/viewHolder/ReportListViewHolder.kt
package com.rendersoncs.reportform.view.adapter.viewHolder

import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface
import android.graphics.drawable.GradientDrawable
import android.view.MenuItem
import android.view.View
import android.widget.PopupMenu
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.rendersoncs.reportform.R
import com.rendersoncs.reportform.itens.ReportItems
import com.rendersoncs.reportform.view.adapter.listener.OnInteractionListener
import com.rendersoncs.reportform.view.services.constants.ReportConstants
import com.rendersoncs.reportform.view.services.extension.StringExtension.limitsText
import kotlinx.android.synthetic.main.activity_main_list.view.*
import java.util.*

class ReportListViewHolder(itemView: View, context: Context) : RecyclerView.ViewHolder(itemView) {
    private var companyView = itemView.companyView
    private var dateView = itemView.dateView
    private var view = itemView.viewResult
    private var resultView = itemView.resultView
    private val overflow = itemView.overflow
    private val mContext = context

    private val result = mContext.getString(R.string.according)
    private val colorAccording = ContextCompat.getColor(mContext, R.color.colorRadioC)
    private val colorNotAccording = ContextCompat.getColor(mContext, R.color.colorRadioNC)

    fun bindData(repoEntity: ReportItems, listener: OnInteractionListener) {

        companyView!!.text = limitsText(repoEntity.company, ReportConstants.CHARACTERS.LIMITS_TEXT)
        dateView!!.text = repoEntity.date
        resultView.text = repoEntity.result

        val getResult = repoEntity.result.toString()

        if (getResult == result.toUpperCase(Locale.ROOT)) {
            changeColorShape(colorAccording)
        } else {
            changeColorShape(colorNotAccording)
        }

        itemView.setOnClickListener { listener.onOpenPdf(repoEntity.id) }
        overflow.setOnClickListener {
            val popupMenu = PopupMenu(mContext, overflow)
            setForceShowIcon(popupMenu)
            popupMenu.inflate(R.menu.menu_main)
            popupMenu.setOnMenuItemClickListener { item: MenuItem ->
                when (item.itemId) {
                    R.id.action_resume -> {
                        listener.onListClick(repoEntity.id)
                        return@setOnMenuItemClickListener true
                    }
                    R.id.action_edit -> {
                        listener.onEditReport(repoEntity.id)
                        return@setOnMenuItemClickListener true
                    }
                    R.id.action_share -> {
                        listener.onShareReport(repoEntity.id)
                        return@setOnMenuItemClickListener true
                    }
                    R.id.action_remove -> {
                        AlertDialog.Builder(mContext)
                                .setTitle(R.string.alert_remove_report)
                                .setMessage(R.string.alert_remove_report_text)
                                .setPositiveButton(R.string.confirm) { _: DialogInterface?, _: Int ->
                                    // Delete item
                                    listener.onDeleteClick(repoEntity.id)
                                }
                                .setNeutralButton(R.string.cancel, null)
                                .show()
                        return@setOnMenuItemClickListener true
                    }
                    else -> return@setOnMenuItemClickListener false
                }
            }
            popupMenu.show()
        }
    }

    private fun changeColorShape(color: Int) {
        val bgShape = view.background as GradientDrawable
        bgShape.setColor(color)
    }

    private fun setForceShowIcon(popupMenu: PopupMenu) {
        try {
            val mFields = popupMenu.javaClass.declaredFields
            for (field in mFields) {
                if ("mPopup" == field.name) {
                    field.isAccessible = true
                    val menuPopupHelper = field[popupMenu]!!
                    val popupHelper = Class.forName(menuPopupHelper.javaClass.name)
                    val mMethods = popupHelper.getMethod("setForceShowIcon",
                            Boolean::class.javaPrimitiveType)
                    mMethods.invoke(menuPopupHelper, true)
                    break
                }
            }
        } catch (e: Throwable) {
            e.printStackTrace()
        }
    }
}